src/Init/Data/Array/Monadic.lean

```diff
@@ -182,11 +182,10 @@ theorem forIn'_eq_foldlM [Monad m] [LawfulMonad m]
   simp [List.forIn'_pure_yield_eq_foldl, List.foldl_map]
 
 @[simp] theorem forIn'_yield_eq_foldl
-    {xs : Array α} (f : (a : α) → a ∈ xs → β → β) (init : β) :
-    forIn' (m := Id) xs init (fun a m b => .yield (f a m b)) =
-      xs.attach.foldl (fun b ⟨a, h⟩ => f a h b) init := by
-  rcases xs with ⟨xs⟩
-  simp [List.foldl_map]
+    {xs : Array α} (f : (a : α) → a ∈ xs → β → Id β) (init : β) :
+    (forIn' xs init (fun a m b => .yield <$> f a m b)).run =
+      xs.attach.foldl (fun b ⟨a, h⟩ => f a h b |>.run) init :=
+  forIn'_pure_yield_eq_foldl _ _
```

Comment:
To elaborate on my comment elsewhere; I think:
* The old lemma  should be kept and deprecated
* The new lemma should be called `idRun_forIn'_yield_eq_foldl` and `simp`

This applies to pretty much every statement modified in this PR.

Reply:
Some existing lemmas follow the naming scheme `id_run_*` (`Array.id_run_foldlM`); others follow the naming scheme `*_id`:

```
@[simp]
theorem findM?_id (p : α → Id Bool) (as : List α) :
    (findM? p as).run = as.find? (p · |>.run) :=
  findM?_pure _ _
```

Should I change these as well? To which naming scheme? The former is more "post-order traversal", the latter is more "in-order traversal". As a user, I think I would have an easier time finding the latter, but of course it depends on whether I have `Id.run *` or `*.run`, so YMMV.

Reply:
That lemma is not an "existing lemma", the `.run` was added in this PR. So I'm pretty convinced we should be adding "run" to these names.

I think the `id_run` name (vs `idRun`) is a pretty weird reading of the naming convention, but since the concern is cosmetic I'll let someone else at the fro decide.

> of course it depends on whether I have Id.run * or *.run, so YMMV.

My guess would be that the former tends to be less confusing, and so in a separate PR it would make sense to tell the delaborator not to use dot notation

Reply:
In conversation with Markus, Paul and Kim, we decided on adopting the `idRun_*` naming convention :+1:

I pushed to #7352 for Kim to take over and will close this PR. Sorry about the chaos :sweat_smile: